source/sigv4.c

```diff
@@ -1199,56 +1194,63 @@ static SigV4Status_t generateCredentialScope( const SigV4Parameters_t * pSigV4Pa
         assert( *canonicalURILen > 0U );
 
         pUriLoc = pUri;
-        pBufLoc = pCanonicalURI;
         bufferLen = *canonicalURILen;
 
-        while( ( i++ < uriLen ) && ( returnStatus == SigV4Success ) )
+        while( ( uriIndex++ < uriLen ) && ( bytesConsumed < bufferLen ) && ( returnStatus == SigV4Success ) )
```

Comment:
I got around this in my MISRA fixes by instead changing the `else if` code from
```c
*pBufLoc = *pUriLoc;
++pBufLoc;
++bytesConsumed;
```
to
```c
if( bytesConsumed < bufferLen )
{
    *pBufLoc = *pUriLoc;
    ++pBufLoc;
    ++bytesConsumed;
}
else
{
    returnStatus = SigV4InsufficientMemory;
    LOG_INSUFFICIENT_MEMORY_ERROR( "encode the URI", bytesConsumed + 1U - bufferLen );
}
```
This appropriately sets the returnStatus to `SigV4InsuffcientMemory` in the loop, instead of adding code later to deal specifically with that case

Reply:
Updated to use the nested if-else approach as complexity isn't a problem anymore